compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -724,13 +725,25 @@ public void visit(BLangPackage pkgNode) {
                 initFnBody.stmts.add(constInit);
             }
         }
+        List<BLangVariable> desugaredGlobalVarList = new ArrayList<>();
 
         pkgNode.globalVars.forEach(globalVar -> {
-            BLangAssignment assignment = createAssignmentStmt(globalVar);
-            if (assignment.expr != null) {
-                initFnBody.stmts.add(assignment);
+            // This will convert complex variables to simple variable
+            if (globalVar.getKind() == NodeKind.TUPLE_VARIABLE) {
+                //Tuple variable will be desugared into block statement node
+                BLangNode blockStatementNode = rewrite(globalVar, env);
+                // Add each desugared simple variable to global variables
+                ((BLangBlockStmt) blockStatementNode).stmts.forEach(bLangStatement -> {
+                    BLangSimpleVariableDef simpleVarDef1 = (BLangSimpleVariableDef) bLangStatement;
+                    addToInitFunction(simpleVarDef1.var, initFnBody);
+                    desugaredGlobalVarList.add(rewrite(simpleVarDef1.var, env));
+                });
+            } else {
+                addToInitFunction((BLangSimpleVariable) globalVar, initFnBody);
+                desugaredGlobalVarList.add(rewrite(globalVar, env));
             }
```

Comment:
Shall we extract this into a separate method?

Reply:
Yes, I moved this into a separate method in my next PR (error and mapping bp support). Let's keep it like this for now